csharp/Microsoft.Azure.Devices.Client.NetMF/AuthenticationMethodFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.Client
+{
+    using System;
+
+    public sealed class AuthenticationMethodFactory
+    {
+        static internal IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionStringBuilder iotHubConnectionStringBuilder)
+        {
+            if (iotHubConnectionStringBuilder.SharedAccessKeyName != null)
+            {
+                //return new DeviceAuthenticationWithSharedAccessPolicyKey(
+                //    iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessKeyName, iotHubConnectionStringBuilder.SharedAccessKey);
+            }
+            else if (iotHubConnectionStringBuilder.SharedAccessKey != null)
+            {
+                return new DeviceAuthenticationWithRegistrySymmetricKey(
+                    iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessKey);
+            }       
+            else if (iotHubConnectionStringBuilder.SharedAccessSignature != null)
+            {
+                return new DeviceAuthenticationWithToken(iotHubConnectionStringBuilder.DeviceId, iotHubConnectionStringBuilder.SharedAccessSignature);
+            }
+
+            throw new InvalidOperationException("Unsupported Authentication Method " + iotHubConnectionStringBuilder.ToString());
+        }
+
+        //public static IAuthenticationMethod CreateAuthenticationWithSharedAccessPolicyKey(string deviceId, string policyName, string key)
```

Comment:
Can you please clarify the underlying reason for removing this method? Is there a missing component on .NETMF forcing to get rid of this?